upup/pkg/fi/cloudup/awstasks/instancerequirements.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awstasks
+
+import (
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+
+	"k8s.io/kops/upup/pkg/fi"
+)
+
+type InstanceRequirements struct {
+	Architecture *string
+	CpuMin       *int64
+	CpuMax       *int64
+	MemoryMin    *int64
+	MemoryMax    *int64
+}
+
+var _ fi.HasDependencies = &InstanceRequirements{}
+
+func (e *InstanceRequirements) GetDependencies(tasks map[string]fi.Task) []fi.Task {
+	return nil
+}
```

Comment:
This doesn't look like an actual task, but some functions that may be best kept in `awstasks/autoscalinggroup.go`.

Reply:
This is how we've done it with other things that should be nested structs. E.g ELB health checks and similar. I have not come across a task that takes in a struct that doesn't implement this.